README.md

```diff
@@ -94,128 +94,118 @@ $ yarn workspace [name-of-package] add [name-of-npm-package]
+First, you'll need to download the `events.csv` from [dropbox](https://www.dropbox.com/s/k86snoabcto42mf/events.csv?dl=0) which is a 22gb file that has all NFT transfers from the first ever block that ERC-721 tokens existed (around block 4.6m).
 
-[//]: # ()
-[//]: # (Once you've got the data file and installed Docker and have the PostgreSQL image running as described above, you'll want to load the data onto the image and persist it to the volume.)
+Once you've got the data file and installed Docker and have the PostgreSQL image running as described above, you'll want to load the data onto the image and persist it to the volume.
 
-[//]: # ()
-[//]: # (*Note that due to the size of the data file, some of these steps might take a while — thus, while developing it might make sense to use a smaller subset of the data.*)
+*Note that due to the size of the data file, some of these steps might take a while — thus, while developing it might make sense to use a smaller subset of the data.*
 
-[//]: # ()
-[//]: # (1. Connect to the Docker image in with your preferred DB client)
+1. Connect to the Docker image in with your preferred DB client
 
-[//]: # ()
-[//]: # (2. Run the following query to create the table / schema:)
+2. Run the server at least once (no dump mode) to get the db and table definitions created
 
-[//]: # ()
-[//]: # (```sql)
-
-[//]: # (CREATE TABLE events &#40;)
-
-[//]: # (    id              INT PRIMARY KEY,)
-
-[//]: # (    block_number    INT,)
-
-[//]: # (    nft_name        TEXT,)
-
-[//]: # (    contract_hash   VARCHAR&#40;66&#41;,)
-
-[//]: # (    txn_hash        VARCHAR&#40;66&#41;,)
-
-[//]: # (    txn_type        VARCHAR&#40;10&#41;,)
-
-[//]: # (    gas             BIGINT,)
-
-[//]: # (    value           VARCHAR&#40;78&#41;,)
-
-[//]: # (    from_hash       VARCHAR&#40;42&#41;,)
-
-[//]: # (    to_hash         VARCHAR&#40;42&#41;,)
-
-[//]: # (    token_id        VARCHAR&#40;78&#41;,)
-
-[//]: # (    timestamp       DATE,)
-
-[//]: # (    createdAt       DATE,)
-
-[//]: # (    updatedAt       DATE)
-
-[//]: # (&#41;;)
-
-[//]: # (```)
-
-[//]: # ()
-[//]: # (3. Then, you'll need to get move the `.csv` file form your host machine to the container.)
-
-[//]: # ()
-[//]: # (    * You can get the container_id of the container by running `docker ps` on the command line)
-
-[//]: # ()
-[//]: # (```bash)
-
-[//]: # ($ docker cp /path/events.csv container_id:/events.csv)
-
-[//]: # (```)
-
-[//]: # ()
-[//]: # (4. Now, populate the Postgres DB by running the following query which copies the `.csv` data into the DB volume)
+```sql
+CREATE TABLE dumped_blocks (
+    id              SERIAL PRIMARY KEY,
+    number          VARCHAR(10),
+    timestamp       DATE,
+    createdAt       DATE,
+    updatedAt       DATE
+);
+```
+```sql
+CREATE TABLE events (
+    id              INT PRIMARY KEY,
+    block_number    VARCHAR(10),
```

Comment:
int

Reply:
I agree that this should be in theory an INTEGER but we had some troubles with this as the value is parsed from any block as string, current code relies on this being an string and we are basically transforming it in js to a number, maybe we can run a migration once we are pulling the data from the blockchain directly ? but otherwise I would not change this as of right now